src/core/statelisteners/searchlistener.js

```diff
@@ -0,0 +1,177 @@
+import { SearchParams } from '../../ui';
+import QueryTriggers from '../models/querytriggers';
+import StorageKeys from '../storage/storagekeys';
+
+/**
+ * SearchListener is responsible for running a vertical or universal search when
+ * the QUERY storage key is updated. It also sets the defaultInitialSearch, if
+ * there is no QUERY already present.
+ */
+export default class SearchListener {
+  constructor (systemConfig, config) {
+    this.storage = systemConfig.storage;
+    this.componentManager = systemConfig.componentManager;
+    this.core = systemConfig.core;
+    this.config = {
+      searchCooldown: 300,
+      verticalKey: undefined,
+      allowEmptySearch: true,
+      defaultInitialSearch: undefined,
+      ...config
+    };
+
+    /**
+     * Middleware functions to be run before a search. Can be either async or sync.
+     */
+    this.middleware = [];
+  }
+
+  /**
+   * Sets the defaultInitialSearch, if no query is already set, and register's the
```

Comment:
Since this function doesn't reference `defaultInitialSearch`, does it make sense to instead have this comment say "Sets the on load query and registers the QUERY listener"?

Reply:
yeah that's better!